src/Tribe/Integrations/WPML/Defaults.php

```diff
@@ -57,116 +47,78 @@ public static function instance() {
-			$translation_management->settings['custom_fields_translation'][ $field ] = WPML_COPY_CUSTOM_FIELD;
+		// if the user already put one in place skip
+		if ( file_exists( $destination ) ) {
+			return true;
 		}
 
-		// remove the method to avoid infinite loops
-		remove_action( 'icl_save_settings', array( $this, 'set_defaults' ) );
+		$written = file_put_contents( $destination, $this->get_config_file_contents() );
 
-		// the Translation Management plugin might not be active on this
-		// installation, save this option only if Translation Management is active.
-		$translation_management->save_settings();
-		Tribe__Settings_Manager::set_option( $this->defaults_option_name, true );
+		if ( ! $written ) {
+			$message = $this->get_config_file_fail_message();
+			$html = sprintf( '<p class="error">%s</p>', esc_html( $message ) );
+			tribe_notice( 'tec-wpml-config-file-not-written', $html, array( 'type' => 'error' ) );
+
+			return false;
+		}
 
 		return true;
 	}
 
 	/**
-	 * Checks whether default custom field translation option values have been for the current installation.
+	 * Returns the path to the WPML config file for the plugin.
 	 *
-	 * @return bool Whether defaaults have been set already or not.
+	 * @return string
 	 */
-	public function has_set_defaults() {
-		return false !== Tribe__Settings_Manager::get_option( $this->defaults_option_name, false );
+	protected function get_config_file_path() {
+		return Tribe__Events__Main::instance()->pluginPath . DIRECTORY_SEPARATOR . 'wpml-config.xml';
 	}
 
 	/**
-	 * Returns a list of The Events Calendar related fields that should be copied in translation by default.
-	 *
-	 * This is not a per-event setting but a per-site setting.
+	 * Returns the default content of WPML config file for the plugin.
 	 *
-	 * @return array
+	 * @return string
 	 */
-	protected function get_default_copy_fields() {
-		$default_copy_fields = array(
-			'_EventAllDay',
-			'_EventStartDate',
-			'_EventEndDate',
-			'_EventStartDateUTC',
-			'_EventEndDateUTC',
-			'_EventDuration',
-			'_EventVenueID',
-			'_EventShowMapLink',
-			'_EventShowMap',
-			'_EventCurrencySymbol',
-			'_EventCurrencyPosition',
-			'_EventCost',
-			'_EventCostMin',
-			'_EventCostMax',
-			'_EventURL',
-			'_EventOrganizerID',
-			'_EventPhone',
-			'_EventHideFromUpcoming',
-			'_EventTimezone',
-			'_EventTimezoneAbbr',
-			'_EventRecurrenceRRULE',
-			'_VenueCountry',
-			'_VenueAddress',
-			'_VenueCity',
-			'_VenueStateProvince',
-			'_VenueState',
-			'_VenueProvince',
-			'_VenueZip',
-			'_VenuePhone',
-			'_VenueURL',
-			'_OrganizerEmail',
-			'_OrganizerWebsite',
-			'_OrganizerPhone',
-		);
-
-		// The reason this array is not filtered is that each plugin should act independently in setting
-		// its own specific defaults, plus this array will be parse once per site and concurrency cannot be
-		// granted.
-		return $default_copy_fields;
+	protected function get_config_file_contents() {
+		return <<< XML
```

Comment:
I am not sure, but can we double check if this is something that works on all servers? I rarely see people use, so it would be interesting to see if there is a way to deactivate this that would throw fatals or warnings on peoples sites.